twoliter/src/cmd/build.rs

```diff
@@ -23,12 +37,102 @@ pub(crate) struct BuildVariant {
     project_path: Option<PathBuf>,
 
     /// The architecture to build for.
-    #[clap(long = "arch", default_value = "x86_64")]
+    #[clap(long = "arch", env = "BUILDSYS_ARCH", default_value = "x86_64")]
     arch: String,
+
+    /// The variant to build.
+    #[clap(env = "BUILDSYS_VARIANT")]
+    variant: String,
+
+    /// The go modules that should be build
+    #[clap(long = "go-modules", env = "GO_MODULES", default_value = "")]
+    go_modules: String,
 }
 
 impl BuildVariant {
     pub(super) async fn run(&self) -> Result<()> {
-        Ok(())
+        let project = project::load_or_find_project(self.project_path.clone()).await?;
+        let tempdir = tools_tempdir()?;
+        install_tools(&tempdir).await?;
+        let makefile_path = tempdir.path().join("Makefile.toml");
+        let packages_dir =
+            TempDir::new().context("Unable to create a tempdir for Twoliter's packages")?;
+
+        let sdk_container = DockerContainer::new(
+            "sdk",
+            project
+                .sdk(&self.arch)
+                .context("The project was missing an sdk")?
+                .uri(),
+        )
+        .await?;
+        sdk_container
+            .cp(Path::new("twoliter/alpha/build/rpms"), packages_dir.path())
+            .await?;
+
+        let rpms_dir = project.project_dir().join("build").join("rpms");
+        fs::create_dir_all(&rpms_dir)?;
+        debug!("Copying rpms to build dir");
+        for maybe_file in fs::read_dir(packages_dir.path().join("rpms"))? {
+            let file = maybe_file?;
+            if !file.file_type()?.is_file() {
+                debug!("Skipping '{}'", file.path().display());
+            }
+            debug!("Copying '{}'", file.path().display());
+            fs::copy(file.path(), rpms_dir.join(file.file_name()))?;
+        }
+
+        // Create the sbkeys directory
+        if !Path::new("sbkeys").is_dir() {
+            create_dir("sbkeys").await?;
+        }
+
+        let created_files = cp_alpha_files(&sdk_container).await?;
+
+        // Hold the result of the cargo make call so we can clean up the project directory first.
+        let res = CargoMake::new(&project, &self.arch)?
+            .env("TWOLITER_TOOLS_DIR", tempdir.path().display().to_string())
+            .env("GO_MODULES", &self.go_modules)
+            .env("BUILDSYS_ARCH", &self.arch)
+            .env("BUILDSYS_VARIANT", &self.variant)
+            .makefile(makefile_path)
+            .project_dir(project.project_dir())
+            .exec("build")
+            .await;
+
+        // Clean up all of the files we created
+        for file_name in created_files {
```

Comment:
This won't run if there's an error somewhere above this. We should try adding these files to a tempdir and then pointing to it. Although... maybe the scripts expect to be in the same dir as the keys that they might use :thinking:

Reply:
The script requires it to be in the sb-keys dir.

Reply:
`twoliter build variant` now uses the existing sbkeys dir, or uses a tempdir if one doesn't exist.